### PR TITLE
feat(exception): migrate exception related source code from TensorBaySDK

### PR DESCRIPTION
**Pull request by Joey9933**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2jn2ppq TensorBay 代码迁移：exception.py](https://app.clickup.com/t/2jn2ppq)

File: graviti/exception.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
```
Comment (Lee-000) [inline graviti/exception.py:3]: 2022？

File: graviti/exception.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    def __init__(self, message: Optional[str] = None, *, is_draft: Optional[bool] = None) -> None:
+        super().__init__()
+        if is_draft is None:
+            self._message = message
+        else:
+            required_status = "commit" if is_draft else "draft"
+            self._message = f"The status is not {required_status}"
+
+
+class DatasetTypeError(ClientError):
```
Comment (Lee-000) [inline graviti/exception.py:51]: Is this still necessary?

File: graviti/exception.py [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    def __str__(self) -> str:
+        if self._dataset_name and self._is_fusion:
+            return (
+                f'Dataset "{self._dataset_name}" is '
+                f'{"" if self._is_fusion else "not "}a fusion dataset'
+            )
+        return super().__str__()
+
+
+class FrameError(ClientError):
```
Comment (Lee-000) [inline graviti/exception.py:80]: Is this still necessary?

File: graviti/exception.py [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+    STATUS_CODE = 500
+
+
+class UnauthorizedError(ResponseError):
+    """This class defines the exception for unauthorized response error."""
+
+    STATUS_CODE = 401
+
+
+class OpenDatasetError(GravitiException):
```
Comment (Lee-000) [inline graviti/exception.py:252]: Is this still necessary?

File: graviti/exception.py [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+            return (
+                f"No module named {self._module_name}."
+                "\n"
+                f'\n    To install the module, please run: "pip3 install {self._package_name}"'
+                "\n"
+            )
+        return super().__str__()
+
+
+class TBRNError(GravitiException):
```
Comment (Lee-000) [inline graviti/exception.py:309]: Is this still necessary?

File: graviti/exception.py [hunk 6]
```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/env python3
```
Comment (Lee-000) [inline graviti/exception.py:1]: The imports in GravitiSDK should also be updated.

**Review by Lee-000 — CHANGES_REQUESTED** [review]
Add the scope in the commit message.

File: graviti/dataset/dataset.py [hunk 7]
```diff
@@ -8,7 +8,6 @@
 from typing import Any, Dict, Iterator, KeysView, Mapping, Optional
```
Comment (linjiX) [inline graviti/dataset/dataset.py:8]: file `openapi/request.py` is just committed, the code should be rebased and update `request.py` too.

Comment (linjiX) [inline graviti/dataset/dataset.py:8]: file `openapi/request.py` just commited, it should be updated too.

Comment (Joey9933) [reply graviti/dataset/dataset.py:8]: Okay

**Review by linjiX — APPROVED** [review]

**Review by graczhual — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**